hangupsbot/plugins/telesync/core.py

```diff
@@ -557,6 +566,34 @@ def _on_membership_change(self, msg):
                 bot.memory.set_by_path(path_chat, 1)
                 type_ = 1
 
+                # check for chat restrictions and apply if necessary
+                restrict = get_sync_config_entry(bot,
+                                                 'telesync:' + msg.chat_id,
+                                                 'restrict_user')
+                rights = (NO_SENDING_RIGHTS if restrict == 'messages' else
+                          NO_MEDIA_RIGHTS if restrict == 'media' else
+                          NO_STICKER_RIGHTS if restrict == 'sticker' else
+                          NO_WEBPREVIEW_RIGHTS if restrict == 'websites' else
+                          NO_WEBPREVIEW_AND_STICKER_RIGHTS if restrict == 'sticker+websites'
+                          else None)
+                if rights and msg['chat']['type'] == 'supergroup':
+                    result = await self.restrictChatMember(msg.chat_id,
```

Comment:
We are actually not in a coroutine here, this requires `async def _on_membership_change(...):` and `await self._on_membership_change(msg)` in line 485.

Reply:
`TelegramBot.restrictChatMember` likely raises an Exception - the membership change would never reach the sync handler in this case.